data-gen/src/main/java/io/smartcat/data/loader/rules/RangeRuleDouble.java

```diff
@@ -17,8 +18,15 @@
     // definition of the range: e.g [a,b,c,d] : a < b <= c < d is a set of ranges: {[a,b),[c,d)}
     private List<Double> ranges = new ArrayList<>();
 
+    private final List<Double> rangeEdges = new LinkedList<>();
+
     private Randomizer random;
 
+    /**
+     * Small value that will be subtracted from the end of ranges.
+     */
+    public static final Double EPSILON = 0.00000000001;
```

Comment:
The problem was that I did not take into account the second subtraction (ten - (ten - eps)) where the approximation was also taking place. The approximation of the result is sometimes slightly below the actual result and sometimes slightly above the actual result, depending of the chosen eps. The test is fixed by using Math.abs(), which checks that the result is close enough to expected.